### ISSUE TITLE
Clean up geometry directories

### ISSUE THREAD

**Issue by wdconinc** [JeffersonLab/remoll#195]
**Is your feature request related to a problem? Please describe.**
There are currently many geometry directories and many entry points into the geometry in those directories. This makes it difficult for new users to determine which geometry is current. Simulations with wrong geometry are wasteful and frustrating.

**Describe the solution you'd like**
We should have a single directory with geometry (or a separate repository) with clear information on what is in which geometry (per README.md in that directory). There should be no duplication of identical geometry elements if it can be at all avoided. Certainly there should be no copying of entire directories.

**Describe alternatives you've considered**
Even if we find ourselves in a geometry mess right now, there is probably useful information in all directories (even if just for historical record keeping). We should try not to throw out directories entirely. Since geometry and code are different things, it might make sense to keep the geometry in a submodules which we can pin at specific revisions and allow independent updates for.

**Comment by wdconinc** [comment]
https://gist.github.com/wdconinc/a1e86476816fdce6f2401ced24095661 may be useful

**Comment by wdconinc** [comment]
We don't just have geometry in master, also in branches. Possibly with changes. Which one of the following branches contain geometry changes (i.e. anything inside geometry*/)  that should be tracked (i.e. merged into develop or so)?

Below branches with name for feedback on any geometry changes (or no name for those where no geometry changes).
-  BeamSteering : no geometry changes
-  GdmlGenerator : @rahmans1 
-  LaTech : @rakithab has geometry changes that need to be figured out
-  bugfix-save-all-hits : @wdconinc geometry changes tracked in PR #197 
-  cadmesh : no geometry changes
-  cameron_dose_branch @cameronc137 lots of geometry changes. which directories are relevant?
-  feature-beam-target-separation : no geometry changes
-  feature-cached-global-field : no geometry changes
-  feature-gamma-tracking : no geometry changes
-  feature-geometry-submodule : @wdconinc changes to geometry structure (merge path)
-  feature-macro-geometry-search-path : no geometry changes
-  feature-store-trajectory : no geometry changes
-  tracking_peri : @cameronc137 geometry changes tracked in closed PR #104
-  vacuum_test : @cipriangal lots of geometry changes.

I don't want to get rid of changes to the geometry in these branches, but it doesn't make sense to track the geometry in a branch that doesn't have any changes anyway. Everything else I think would go to a remoll-geometry submodule.

**Comment by rahmans1** [comment]
@wdconinc I am not making any changes to geometry in the beam steering branch. The GdmlGenerator branch seems to be ancient and should probably be deleted. We are tracking it separately for now in its own repo. We can keep doing that or create a new feature branch to track the GdmlGenerator starting with a more recent iteration. What do you think @cameron

**Comment by cipriangal** [comment]
The vacuum_test does have changes that will need to merged into develop and changes that will probably be best to remove. My plan is to finish up some studies this week (we have a meeting coming up on Th with Thia), merge the whole branch into develop and then remove the things that I think are useless.

**Comment by wdconinc** [comment]
@cipriangal There's no rushed needed for merging things in, necessarily. I do want the default develop to be a realistic geometry by the end of the week. If you think your changes can wait until the next release, that's fine too.

**Comment by wdconinc** [comment]
Also, that whole exercise of excising the geometry into remoll-geometry results in a large number of unmergeable branches (check out JeffersonLab/remoll-geometry). That avenue is not going to be helpful in merging ourselves out of this mess. Still, keeping the geometry in a different place is probably the right thing to do.

**Comment by cameronc137** [comment]
The cameron_dose_branch is old and can be deleted now (it is tracked in a fork). The tracking_peri branch has been superseded by Catherine's updates with trajectories and Daniel's updates with analysis scripts in the analysis folder, both are up to date in develop.

**Comment by wdconinc** [comment]
Ok, can you go ahead and delete them? And then focus on those directories in develop that you know the status of...

**Comment by cameronc137** [comment]
Yes, but it may take a few days, as I'm about to be the APEX RC. All geometry_*/ folders are defunct or managed in more recent parallel branches as far as I know.

**Comment by cameronc137** [comment]
We can completely delete the geometry generator and tracking_peri branches

**Comment by wdconinc** [comment]
Current geometry status...
![geometry](https://user-images.githubusercontent.com/4656391/53125050-f82b5a00-352a-11e9-8d2c-6cfeda1f41bb.png)

**Comment by wdconinc** [comment]
Apparently I reinvented the wheel again and forgot that this is already in the repository... https://github.com/JeffersonLab/remoll/blob/master/scripts/dot.sh

**Comment by wdconinc** [comment]
Branch https://github.com/JeffersonLab/remoll/tree/geometry-pruning should be merged into develop for this.

**Comment by wdconinc** [comment]
So, now in develop we've got a more cleaned up version:
```
geometry
├── detector_5open.gdml
├── detector_mainz.gdml
├── detector_parametrized.gdml
├── dumpDaughter.gdml
├── genquartz.py
├── hall
│   ├── hallDaughter_acceptanceDefinition.gdml
│   ├── hallDaughter_dump.gdml
│   ├── hallDaughter_merged.gdml
│   ├── hallDaughter_noShlds.gdml
│   └── schema -> ../schema
├── hybrid
│   ├── hybridDaughter_acceptanceDefinition.gdml
│   ├── hybridDaughter_acceptanceDefinition_includeCol4.gdml
│   ├── hybridDaughter.gdml
│   ├── hybridDaughter_merged.gdml
│   ├── hybridDaughter_noShlds.gdml
│   └── schema -> ../schema
├── issues
│   └── issue172.gdml
├── materials.xml
├── matrices.xml
├── misc
│   ├── schema -> ../schema
│   └── surveymarker.gdml
├── mollerMother_5open.gdml
├── mollerMother_acceptanceDefinition.gdml
├── mollerMother_acceptanceDefinition_includeCol4.gdml
├── mollerMother_black.gdml
├── mollerMother_Clamshell_Optimized.gdml
├── mollerMother_dump.gdml
├── mollerMother.gdml
├── mollerMother_merged.gdml
├── mollerMother_noShlds.gdml
├── mollerMother_parametrized.gdml.disabled
├── mollerParallel.gdml
├── pion
│   ├── Calorimeter
│   │   ├── pionCalorimeterMaterials.xml
│   │   ├── pionCalorimeterMatrices.xml
│   │   ├── pionDetectorCalorimeter.gdml
│   │   └── schema -> ../schema
│   ├── GEM
│   │   ├── pionDetectorGEM.gdml
│   │   ├── pionDetectorGEMOpenSector.gdml
│   │   └── schema -> ../schema
│   ├── Lucite
│   │   ├── pionDetectorLucite.gdml
│   │   ├── pionDetectorLuciteSector.gdml
│   │   ├── pionDetectorLucite_world.gdml
│   │   ├── pionLuciteMaterials.xml
│   │   ├── pionLuciteMatrices.xml
│   │   └── schema -> ../schema
│   ├── pionDetectorSystem.gdml
│   ├── pionDetectorVirtualDet.gdml
│   ├── schema -> ../schema
│   └── TS
│       ├── pionDetectorTS.gdml
│       ├── pionDetectorTSOpenSector.gdml
│       └── schema -> ../schema
├── positions.xml
├── quartzdet.py
├── schema -> ../schema
├── showermax
│   ├── schema -> ../schema
│   ├── showerMaxDetector.gdml
│   ├── showerMaxDetectorSystem.gdml
│   ├── showerMaxMaterials.xml
│   └── showerMaxMatrices.xml
├── solids
│   └── world.xml
├── solids_5open.xml
├── solids_mainz.xml
├── target
│   ├── schema -> ../schema
│   ├── subTargetRegion.gdml
│   ├── target12C.gdml
│   ├── targetDaughter_acceptanceDefinition.gdml
│   ├── targetDaughter_Clamshell_Optimized.gdml
│   ├── targetDaughter.gdml
│   ├── targetDaughter_merged.gdml
│   └── targetDaughter_noShlds.gdml
├── tracking
│   ├── daughterGEM1.gdml
│   ├── daughterGEM2.gdml
│   ├── daughterGEM3.gdml
│   ├── daughterGEM4.gdml
│   ├── schema -> ../schema
│   └── trackingDaughter.gdml
└── upstream
    ├── schema -> ../schema
    ├── upstreamDaughter_acceptanceDefinition.gdml
    ├── upstreamDaughter.gdml
    ├── upstreamDaughter_merged.gdml
    └── upstreamDaughter_noShlds.gdml
```
Not sure what to do with the python scripts.

**Comment by wdconinc** [comment]
I don't like the schema links all over, but at least in the short term it's a pain worth bearing. Maybe we can store them at moller.jlab.org/gdml, which is accessible from the farm. The main reason for going with a local schema copy was that we couldn't access the CERN gdml schema link from the farm. Unfortunately it doesn't seem possible with geant4 to set your own local schema search path (or I haven't found out how).

**Comment by wdconinc** [comment]
`hall/hallDaughter_acceptanceDefinition.gdml == hall/hallDaughter_noShlds.gdml` too.

**Comment by wdconinc** [comment]
Enough on this for v2.1, bumping to v2.2
